spreadsheets/spread-sheet.ts

```diff
@@ -0,0 +1,77 @@
+import Sheets from 'node-sheets';
+
+export class SpreadSheet {
+  public static sheetTabularData = {};
+  private static lastUpdate = null;
+  private static currUpdateTime = null;
+  private static googleSheet: Sheets;
+  private static sheetAuthData: any;
+
+  private constructor() {}
+
+  /**
+   * Initialize the spreadsheet
+   * @param sheetId
+   * @param authData
+   */
+  public static initSpreadSheet(sheetId: string, authData: any) {
+    SpreadSheet.googleSheet = new Sheets(sheetId);
+    SpreadSheet.sheetAuthData = authData;
+  }
+
+  /**
+   * Check for updates in the spreadsheet. Returns a Promise that resolves to a boolean.
+   */
+  public static checkForUpdate = async (): Promise<boolean> => {
+    try {
+      await SpreadSheet.authorizeSheet();
+      const date = new Date(await SpreadSheet.googleSheet.getLastUpdateDate()).getTime();
+      SpreadSheet.currUpdateTime = date;
+      return !SpreadSheet.lastUpdate ? true : SpreadSheet.lastUpdate < date;
+    } catch (e) {
+      console.error(e);
+      return false;
+    }
+  };
+
+  /**
+   * Stores the tabular content of each sheet in the object sheetTabularData with the sheet name as the key.
+   */
+  public static fetchUpdatesFromGoogleSheet = async () => {
+    try {
+      await SpreadSheet.authorizeSheet();
+      const sheetNames = await SpreadSheet.getSheetNames();
+
+      sheetNames.forEach(async tableName => {
+        const table = await SpreadSheet.googleSheet.tables(tableName);
```

Comment:
I think this will be called _sequentially_, meaning each sheet will wait for the next sheet call to complete before it starts.

An improvement here will be to fetch them in parallel. We can add that as an improvement GH issue later, if you don't want to deal with it now, it's fine

Reply:
Updated this to use **Promise.all()**. Please review